src/fiber/fiber.ml

```diff
@@ -427,10 +431,7 @@ let finalize f ~finally =
   in
   match res with
   | Ok x -> return x
-  | Error l ->
-    let* () = parallel_iter l ~f:(fun exn -> Exn_with_backtrace.reraise exn) in
-    (* We might reach this point if all raised errors were handled by the user *)
```

Comment:
Was this comment wrong, then?

Reply:
It's right, but we handle this case explicitly: `[] -> never`.